features/step_definitions/features_steps.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+Given('a feature {string} of {provider}') do |feature_name, provider|
+  @service_plan = provider.service_plans.where(name: "Basic").last
+  feature = @service_plan.service.features.find_or_create_by(name: feature_name, scope: "ServicePlan")
+end
+
+Given('an enabled feature {string} of {provider}') do |feature_name, provider|
+  @service_plan = provider.service_plans.where(name: "Basic").last
+  feature = @service_plan.service.features.find_or_create_by(name: feature_name, scope: "ServicePlan")
+  @service_plan.features << feature
+  @service_plan.save!
+end
+
+Then('there {is} no feature named {string}') do |is, string|  
+  feature_name = @service_plan.service.features.last.name
+  assert_not_equal feature_name, string
+end
+
+Then('there {is} feature named {string}') do |is, string|
+  feature_name = @service_plan.service.features.last.name
+  assert_equal feature_name, string
+end
+
+When("I {string} the feature {string}") do |enable_disable, feature_name|
+  feature = @service_plan.service.features.find_by(name: feature_name)
```

Comment:
This step should not assume that the feature belongs to service, and we should not know its ID. We are trying to replicate the users steps. The user will just see a table with different features in it, will find the row with the desired feature and click on the icon in that row. They will never care about the ID - it is all transparent.

So, the step should just do that - find the feature in the table by the name and click the icon in that row.